src/Http/WebUtilities/src/FileBufferingReadStream.cs

```diff
@@ -208,39 +208,41 @@ private Stream CreateTempFile()
                 FileOptions.Asynchronous | FileOptions.DeleteOnClose | FileOptions.SequentialScan);
         }
 
-        public override int Read(byte[] buffer, int offset, int count)
+        public override int Read(Span<byte> buffer)
```

Comment:
I wonder if we shouldn't do something like the runtime does with [its Sync/AsyncReadWriteAdapter](https://github.com/dotnet/runtime/blob/a4c4efcdb8bb3a30e064bb57e617866f367f8e52/src/libraries/System.Net.Security/src/System/Net/Security/ReadWriteAdapter.cs) to avoid duplicated this somewhat complicated logic. Seems like that way we'd be a lot less likely to introduce bugs in just one version of Read(Async).